gapic/src/main/com/google/gapid/perfetto/canvas/Panel.java

```diff
@@ -148,6 +149,36 @@ public boolean click() {
         }
       };
     }
+
+    public default Panel.Hover withClick(BooleanSupplier onClick) {
+      return new Hover() {
+        @Override
+        public Area getRedraw() {
```

Comment:
[Possibly ignorant question]: I'm wondering, is the overridden function doing anything different?

Reply:
`withClick` creates a delegate that augments the `click()` function below. However, all other calls have to be propagated to the delegate.